### PR TITLE
Adding script for setting up Telepresence for local development

### PR DESCRIPTION
**Pull request by rhuss**
**What this PR does / why we need it**:

This PR helps set up [Telepresence](https://www.telepresence.io/) for allowing local testing and debugging of `cmd/manager/main.go` directly from the IDE. Telepresence takes care for setting up a VPN that allows bi-directional communication, i.e. so that the local process can act as a webhook for the remote cluster. This allows for very quick _inner loop_ development with short turnaround times as you don't need to create container images and redeployments.

Usage:

```
# Install everything. This command is idempotent and can
# be called multiple times
hack/telepresence-setup.sh

# Remove everything from telepresence
hack/telepresence-setup.sh uninstall
```
 
This script has been tested for a KServe installation on Minikube, running on macOS 13.5

More information for using Telepresence for developing:

 * https://codefresh.io/blog/telepresence-2-local-development/
 * https://ttt.io/kubernetes-admission-controller

### CODE REVIEW

File: hack/telepresence-setup.sh
Comment (matzew) [inline hack/telepresence-setup.sh]: perhaps pointing to here: https://www.getambassador.io/docs/telepresence/latest/install

Since `brew` is a Mac specific tool?

Comment (rhuss) [reply hack/telepresence-setup.sh]: Good point, will update. Thanks!

File: hack/telepresence-setup.sh
Comment (matzew) [inline hack/telepresence-setup.sh]: and windows? :smile:

Comment (rhuss) [reply hack/telepresence-setup.sh]: ok, removed Linux :-)

Comment (rhuss) [reply hack/telepresence-setup.sh]: but I doubt that this script will work for Windows anyways ....

Comment (matzew) [reply hack/telepresence-setup.sh]: but `Telepresence` offers downloads for that platform :-)

**Review by terrytangyuan — COMMENTED** [review]
This would be a great addition. There's another alternative https://github.com/txn2/kubefwd but it didn't work well for me. 

For this PR, it would be great if we could add some CI/test that uses this Telepresence setup. It would be great to update contributing guide as well.

**Comment by rhuss** [comment]
@terrytangyuan, do you mean a script that runs the CI tests locally, with the local code, against a cluster tunneled by Telepresence? That sounds like a great idea, so developing CI tests might become faster with faster turnarounds.

In the meantime, I think merging this PR is an excellent first step and harmless, as it just adds some help when developing operators and especially webhooks locally that are called from within the cluster. My primary motivation is to allow debugging by setting breakpoints directly from the IDE and quick turnaround times without building container images (just recompiling within the IDE)

**Comment by terrytangyuan** [comment]
Yep that's exactly the idea.

I'll leave this to reviewers/approvers to comment.

**Comment by rhuss** [comment]
If there is any interest, I could record a demo with a development workflow that involves this script. Just ping me in the case and I will add it to this issue.

**Review by terrytangyuan — APPROVED** [review]
/lgtm

**Pull request merged**
